Make supplier result pagination parameterisable

sarneaud opened the pull request.

matt-sm commented:
What problem is this trying to solve?

sarneaud commented:
The 100-per-page setting is useful if you actually want that, otherwise it just adds complexity (and if anything makes things less efficient).

This morning I found and had to fix two bad supplier data instances in the database.  I decided to write a tool to help with that, but the fixed pagination of /suppliers got in the way.

100-per-page is really designed for one specific application - pagination on the frontend.  However, if we want to change the pagination on the frontend, we should just be able to change it there instead of updating the API server.

matt-sm commented on app/utils.py:
```diff
@@ -47,6 +47,16 @@ def get_int_or_400(data, key):
+def get_nonnegative_int_or_400(data, key, default=None):
+    try:
+        value = int(data.get(key, default))
+        if value < 0:
```
I think it would make sense if size=0 also returned 400

`if value <= 0`

sarneaud commented:
I updated the API a little.  "size" was copied from the search result API, but it didn't make so much sense for pagination, so I changed it to "per_page".  Also, the original idea was that size=0 would remove the limit, but that's really not intuitive, either.  Instead I'll make the frontend code set the per_page parameter where it wants pagination, and just not set a default in the API.
